packages/components/src/sites-table/use-sites-table-sorting.tsx

```diff
@@ -35,24 +38,63 @@ export function useSitesTableSorting< T extends SiteDetailsForSorting >(
 	}, [ allSites, sortKey, sortOrder ] );
 }
 
-function sortSitesAlphabetically< T extends SiteDetailsForSorting >(
+function sortSitesByLastInteractedWith< T extends SiteDetailsForSorting >(
 	sites: T[],
 	sortOrder: SitesTableSortOrder
-): T[] {
-	return [ ...sites ].sort( ( a, b ) => {
-		const normalizedA = a.name.toLocaleLowerCase();
-		const normalizedB = b.name.toLocaleLowerCase();
+) {
+	const interactedItems = sites.filter(
+		( site ) => site.user_interactions && site.user_interactions.length > 0
+	);
+	const remainingItems = sites.filter( ( site ) => site.user_interactions?.length === 0 );
 
-		if ( normalizedA > normalizedB ) {
-			return sortOrder === 'asc' ? 1 : -1;
-		}
+	return [
+		...interactedItems.sort( ( a, b ) => {
+			if ( ! a.user_interactions || ! b.user_interactions ) {
+				return 0;
+			}
 
-		if ( normalizedA < normalizedB ) {
-			return sortOrder === 'asc' ? -1 : 1;
-		}
+			// Interactions are sorted in descending order.
+			const lastInteractionA = a.user_interactions[ 0 ];
+			const lastInteractionB = b.user_interactions[ 0 ];
 
-		return 0;
-	} );
+			if ( new Date( lastInteractionA ) > new Date( lastInteractionB ) ) {
```

Comment:
As we store dates in `2022-09-13` format, would it be enough to compare strings instead of `Date()` objects?

Reply:
It would, but I chose to make it sounder by comparing dates instead of relying on string comparison. For me, at least, it's not really obvious that we're comparing dates considering `lastInteraction` is a string. I think wrapping it's more idiomatic to wrap it within a `Date` object. But let me know your thoughts!

Reply:
I haven't checked the performance difference, so my only argument here was to point out that function call is possibly unnecessary.

If we decide to leave `new Date()` calls, it would make sense to add them in the other sorting function to keep it consistent.

Reply:
That's fair, too. If the `updated_at` field follows the same date signature and it's working, I might as well remove it from here. I'll check and take according action.

Reply:
Update: removed the wrappers, it works even with the full date including time. Thanks, @wojtekn! TIL!